### PR TITLE
Fix issue with GNA Import/Export.

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
- Application may create ostream and write some app-specific header to beginning of stream.
GNA plug-in shall not 'seekg' to beginning of stream where app-specific content is located.
Instead it shall seek to the original position of stream
- Functional test is also updated to cover this case

### Tickets:
 - 50239

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_model_serial.cpp [hunk 1]
```diff
@@ -77,12 +77,13 @@ const int gna_header_magic = is_little_endian() ?  0x4d414e47 : 0x474e414d;
 
 GNAPluginNS::HeaderLatest::ModelHeader GNAModelSerial::ReadHeader(std::istream &is) {
     is.exceptions(std::istream::failbit);
+    auto startPos = is.tellg();
     is.seekg(0, is.end);
-    auto stream_len = is.tellg();
+    auto stream_len = is.tellg() - startPos;
     if (stream_len == -1) {
```
Comment (nkogteva) [inline inference-engine/src/gna_plugin/gna_model_serial.cpp:83]: I suppose this check should be moved higher

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/import_export_tests/import_reshape_permute_conv.cpp [hunk 2]
```diff
@@ -14,13 +14,23 @@ namespace {
 class ImportReshapePermuteConvGNA : public ImportReshapePermuteConv {
 private:
     void exportImportNetwork() override {
-        executableNetwork.Export(fileName);
-        std::fstream inputStream(fileName, std::ios_base::in | std::ios_base::binary);
-        if (inputStream.fail()) {
-            FAIL() << "Cannot open file to import model: " << fileName;
+        static const char appHeader[] = "APPLICATION_HEADER";
```
Comment (nkogteva) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/import_export_tests/import_reshape_permute_conv.cpp:17]: It is better to parameterize header presence

Comment (nosovmik) [reply inference-engine/tests/functional/plugin/gna/shared_tests_instances/import_export_tests/import_reshape_permute_conv.cpp]: Done. But in this case I had to also change base ImportNetworkTestBase class which affected also "Myriad" plugin tests. But now base implementation also has possibility to verify import/export with app headers

File: inference-engine/src/gna_plugin/gna_model_serial.cpp (hunk 1 above)
Comment (nkogteva) [reply inference-engine/src/gna_plugin/gna_model_serial.cpp:83]: I meant that there is a sense to check startPos at first

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/import_export_tests/import_reshape_permute_conv.cpp [hunk 3]
```diff
@@ -14,13 +14,23 @@ namespace {
 class ImportReshapePermuteConvGNA : public ImportReshapePermuteConv {
 private:
     void exportImportNetwork() override {
-        executableNetwork.Export(fileName);
-        std::fstream inputStream(fileName, std::ios_base::in | std::ios_base::binary);
-        if (inputStream.fail()) {
-            FAIL() << "Cannot open file to import model: " << fileName;
+        {
```
Comment (nkogteva) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/import_export_tests/import_reshape_permute_conv.cpp:17]: style

File: inference-engine/src/gna_plugin/gna_model_serial.cpp (hunk 1 above)
Comment (nosovmik) [reply inference-engine/src/gna_plugin/gna_model_serial.cpp:86]: Fixed. Unit test was also implemented for this

**Review by nkogteva — APPROVED** [review]

**Pull request merged**